src/grid_interpolator.cc

```diff
@@ -0,0 +1,104 @@
+#include "flyft/grid_interpolator.h"
+
+#include <fstream> 
+#include <cassert>
+
+namespace flyft
+{
+GridInterpolator::GridInterpolator(std::string s)
+    :GridInterpolator(s, 0, 0, 0)
+    {
+    // Default constructor for the ThreeDimensionIndex
+    }
+
+GridInterpolator::GridInterpolator(std::string s, int ni, int nj, int nk):
+    n_(ni, nj ,nk)
+    {
+    double dx, dy, dz;
+    std::ifstream indata;
+    indata.open(s);
+    if(!indata.is_open())
+        {
+        std::cout<<"File failed to open"<<std::endl;
+        }
+    indata>>ni>>nj>>nk>>dx>>dy>>dz;
+    indata.close();
+    GridInterpolator(s, ni, nj, nk, dx, dy, dz);
+    }    
+
+GridInterpolator::GridInterpolator(std::string s, int ni, int nj, int nk, double dx, double dy, double dz):
+    n_(ni, nj, nk), dx_(dx), dy_(dy), dz_(dz)
+    {
+    data_ = new double[n_.size()];  
+    std::ifstream indata;
+    indata.open(s);
+    while(indata.good()) 
+        {
+        double value;
+        // Ignore the first line of the input file
+        indata.ignore(6,'\n');
+        while(indata >> value)
+            {
+            for (int i = 0; i < ni; i++)
+                {
+                for (int j = 0; j < nj; j++)
+                    {
+                    for (int k = 0; k < nk; k++)
+                        {
+                        data_[n_(i, j, k)] = value;
+                        }
+                    }
+                }
+            }
```

Comment:
This is for sure not right, you are looping over every index in the data and assigning it value each time. So by the time this is done, the whole array will have been iterated n_.size() times, but every entry will be the last value!

It should be more like this:
```suggestion
        int idx = 0;
        while (indata >> value)
            {
            data_[idx++] = value;
            }
       if (idx != n_.size() || !indata.eof())
           {
           throw std::runtime_error("File did not have right number of elements.");
           }
```